xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -1004,7 +1004,7 @@ int CDecoder::Decode(AVCodecContext* avctx, AVFrame* frame)
     return VC_BUFFER;
   }
   else
-    return 0;
+    return VC_BUFFER;
```

Comment:
yes, currently that would not change behaviour but mix-up things somehow. requesting a buffer for the drain path is not exactly the same thing as requesting a buffer after a pic was decoded.